### PR TITLE
Cookie banner and cookie consent (#24)

### PR DESCRIPTION
**Pull request by giulio-giunta**
Hi guys, 
please check my work.
Giulio

### CODE REVIEW

File: web/settings.py [hunk 1]
```diff
@@ -131,3 +133,56 @@
...
+            ],
+        },
+        {
+            "id": "analytics",
+            "name": _("Analytics"),
+            "optional": True,
+            "cookies": [
+                {
+                    "pattern": "_pk_.*",
+                    "description": _("Matomo cookie for website analysis."),
```
Comment (jamesgeddes) [inline web/settings.py:183]: Are we using Matomo now?

**Comment by Freshrojek** [comment]
Hey James!
I'm not sure sorry I haven't done anything to do with the cookies yet, I'm
gonna work on the recaptcha more tonight.
See you soon,
Jack

**Review by CarwynNelson — CHANGES_REQUESTED** [review]
I have pulled this down on to my local machine and I did not get a cookie prompt, which makes sense as there have been no changes to the templates.

It seems that the docs for this library are not very good as it does not mention that we need to include a few extra things in a base template to get this working. The docs also neglect to mention a dependency on `jquery` and `jquery-modal` which I am not a fan of at all.

I managed to get this running locally and I will detail the steps if you are interested to see it working, but after some further research I think we would be better off doing the cookie consent banner ourselves. This is not as scary as it might seem, and I'm more than happy to provide you with guidance to help you get it implemented (there are several avenues we can take).

In order to get this running locally:
1. Add `{% load cookiebanner %}` after line 1 in `memberships/templates/base.html`
2. Add the following after the main css file is loaded in that same base.html
```html
<script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha256-4+XzXVhsDmqanXGHaHvgh1gMQKX40OUvDEBTu8JcmNs=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.css" />
```
3. After the body tag in that same template file add `{% cookiebanner_modal %}`

After doing that and running the django development server you should be presented with something like this on first page load.

image

**Comment by jamesgeddes** [comment]
> I'm not sure sorry I haven't done anything to do with the cookies yet, I'm

This is a request to review the code that @giulio-giunta has submitted, so you would take a look and add any comments you feel necessary :)

**Comment by jamesgeddes** [comment]
Closed by mistake, soz @giulio-giunta !

**Comment by giulio-giunta** [comment]
Hi guys, 
I adapted the cookie banner and removed a lot of stuff to best fit it in the page. I was not able to make work the buttons to save and accept cookies. Sorry!!!
I will need some help for that as it involves JavaScript. If possible I would like to assist who fixes it so I can understand the logic. 
Many thanks.
Giulio

**Pull request closed without merging**